test/can-place-dep.js

```diff
@@ -0,0 +1,1366 @@
+process.env.ARBORIST_DEBUG = '1'
+const t = require('tap')
+const CanPlaceDep = require('../lib/can-place-dep.js')
+const {
+  CONFLICT,
+  OK,
+  REPLACE,
+  KEEP,
+} = CanPlaceDep
+
+const Node = require('../lib/node.js')
+
+t.test('basic placement check tests', t => {
+  const path = '/some/path'
+
+  // boilerplate so we can define a bunch of test cases declaratively
+  const runTest = (desc, {
+    tree,
+    targetLoc,
+    nodeLoc,
+    dep,
+    expect,
+    expectSelf,
+    preferDedupe,
+    peerSet,
+    explicitRequest,
+  }) => {
```

Comment:
i really like how expressive this wrapper makes the tests, but i think it may be useful to put some comments here that explain exactly what each property is for so our future selves don't have to dig too far to remember how to add a new test

Reply:
Good call, will do.

Reply:
https://github.com/npm/arborist/commit/9c53bb310b4a650c8fd9b11413a611b7672ce97c